org.eclipse.jdt.core/dom/org/eclipse/jdt/internal/core/dom/rewrite/ASTRewriteFlattener.java

```diff
@@ -919,7 +919,7 @@ public boolean visit(RecordDeclaration node) {
 		visitList(node, RecordDeclaration.SUPER_INTERFACE_TYPES_PROPERTY, String.valueOf(','), "implements ", Util.EMPTY_STRING); //$NON-NLS-1$
 
 		this.result.append('{');
-		visitList(node, RecordDeclaration.BODY_DECLARATIONS_PROPERTY, Util.EMPTY_STRING, String.valueOf(';'), Util.EMPTY_STRING);
+		visitList(node, RecordDeclaration.BODY_DECLARATIONS_PROPERTY, Util.EMPTY_STRING, Util.EMPTY_STRING, Util.EMPTY_STRING);
```

Comment:
Why don't we use 	this - visitList(ASTNode parent, StructuralPropertyDescriptor childProperty, String separator) - method since the last params are all EMPTY/NULL?

Reply:
I have updated it with the requested change